### ISSUE TITLE
Ability to use custom QueryParsers

### ISSUE THREAD

**Issue by dustinschultz** [DmitryKey/luke#25]
Hi,

Feature Request

There's already the ability to have custom analyzers but it'd be really nice if you could configure a custom QueryParser as well. Custom functionality often requires subclassing QueryParser and not being able to use it in Luke means resorting to not using Luke at all.

In addition, there are several supported versions of subclasses of QueryParser that are bundled with lucene (e.g. AnalyzingQueryParser)

Thanks,
Dustin

**Comment by DmitryKey** [comment]
good point @dustinschultz . Luke allows for custom analyzers. See: https://github.com/DmitryKey/luke/blob/master/luke.sh

Adding support for query parsers sounds logical. Will need to poke around the code and see how to do this.

**Comment by Moddus** [comment]
Hi everyone,
i've tried to implement the functionality to specify a custom QueryParser.
In the code I've found the Parameter '-xmlQueryParserFactory' but I've no
idea how this parameter works. Could anyone give me some advice?

Thanks

**Comment by DmitryKey** [comment]
hi @Moddus !

Tracing the code, I see that it is the xml parser in lucene. Check the org.apache.lucene.queryparser.xml. Take a look at wiki:https://cwiki.apache.org/confluence/display/solr/Other+Parsers#OtherParsers-XMLQueryParser

If you don't pass the command line option you mentioned, the default parser is going to be used: CorePlusExtensionsParser.

**Comment by Moddus** [comment]
Thanks @DmitryKey for your advice. 
I'm looking forward to make the QueryParser configurable.

**Comment by DmitryKey** [comment]
Exciting, thanks @Moddus

**Comment by mocobeta** [comment]
For contributors: [Now Luke is a part of Apache Lucene](https://medium.com/@mocobeta/luke-become-an-apache-lucene-module-as-of-lucene-8-1-7d139c998b2). Please create an issue (if there is not exist) to [Lucene Jira](https://issues.apache.org/jira/projects/LUCENE/summary) if you want to help with this.
